### PR TITLE
Add "Needs Response" disputes page filter option

### LINKED ISSUE #4124: Disputes page: Display "Needs response" items by default
### Description

In the disputes page, we display "Needs response" items by default rather than all items.

@dechov mentioned this:

> It’s now possible with the “advanced filters”, but maybe we want a shortcut to a view that filters to “needs response” and sorts by due date?

@reykjalin: 

> We probably should default to showing the disputes that need a response.

### Acceptance criteria

- Add "Needs Response" filter to the disputes "Show" filter that can be selected on page load using a `filter` query param. (#4361)
- Selecting this filter option will only show disputes with one of the following statuses: `needs_response` or `warning_needs_response`. (#4361) 
- This filter should be selected by default if one or more disputes are awaiting a response.

### Designs

image

### Additional context

p1649724071565769-slack-CGGCLBN58
pdjTHR-17I-p2#comment-1440

cc @LevinMedia

### PR DESCRIPTION
**Pull request by Jinksi**
Partially fixes #4124

#### Changes proposed in this Pull Request

* A "Needs response" filter is added to the "Show" dropdown on the **Payments → Disputes** page.
* When selected, disputes with the status `needs_response` or `warning_needs_response` are displayed.
* This filter can be selected on page load if the `filter=awaiting_response` query param is provided.

image

Filtering by multiple dispute `status` values is achieved with the WCPay-server disputes API `search` query param added in 2127-gh-Automattic/woocommerce-payments-server

#### Testing instructions

1. Checkout the WCPay-server branch: `add/2124-disputes-endpoints-allow-filtering-by-multiple-status` (2127-gh-Automattic/woocommerce-payments-server) and start the local WCPay-server environment with `npm start`.
2. Checkout this WCPay client branch and build JS bundles with `npm run build:client` or `npm start`
3. In your local WCPay (client) store, create disputed purchases with various statuses, including [`needs_response`, `warning_needs_response`, `lost`]:
   * Create a `warning_needs_response` disputed purchase with the card: `4000000000005423`.
   * Create a `needs_response` disputed purchase with the card: `4000000000000259`.
   * Create a `lost` disputed purchase with the card: `4000000000000259` and **Accept** the dispute as merchant.
4. View the **Payments → Disputes** page in wp-admin.
5. The "Show" dropdown should have "All disputes" selected and include the "Needs response" option at the top:
   image
6. Selecting the "Needs response" option will:
   - trigger new network GET requests: `disputes` and `disputes/summary`
   - once resolved, the disputes table will only show disputes needing a response (*Needs Response* or *Inquiry: Needs Response*)
7. Manually **removing** the `filter=awaiting_response` query param from the browser URL bar and hitting `return` will select the "All disputes" filter on page load.
8. Manually **adding** the `filter=awaiting_response` query param to the browser URL bar and hitting `return` will select the "Needs response filter on page load.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by Jinksi** [comment]
I've removed the goal of selecting the filter by default from this PR (code removed in 6bffb67).

Rather than showing the `needs_response` filter by default, we can use the logic used for the disputes notification badge (#4326) to add the `filter=needs_response` param to the Payments → Disputes link URL.

File: client/disputes/filters/config.ts [hunk 1]
```diff
@@ -70,6 +75,10 @@ export const filters: [ DisputesFilterType, DisputesFilterType ] = [
 		],
 		showFilters: () => true,
 		filters: [
+			{
+				label: __( 'Needs response', 'woocommerce-payments' ),
+				value: 'awaiting_response',
```
Comment (Jinksi) [inline client/disputes/filters/config.ts:80]: I'm using the value `awaiting_response` to help decouple the meaning of this filter from the dispute statuses `needs_response` and `warning_needs_response`.

**Comment by james-allan** [comment]
This looks good to me. 🏅 💯 

I've been doing a lot of dispute testing locally and so I had 26 disputes which are in need of a response. This actually came in handy here because I could test paging and changing the number of disputes per page etc. All seemed to work well. 

Screen Shot 2022-06-22 at 6 05 40 pm

Screen Shot 2022-06-22 at 6 06 00 pm

I have a few Lost disputes too and when querying for "All", they all appeared.

**Review by brucealdridge — APPROVED** [review]
LGTM, have tested with a few different cards and correct disputes show when `Needs Response` is selected.

**Comment by Jinksi** [comment]
Note to reviewers: I've updated/added JS tests in 6575e34 and 48f8f9d to cover the `awaiting_response` filter param

**Comment by Jinksi** [comment]
@james-allan I just want to confirm that you're happy with this before I hit merge. I've added/updated some tests since [your comment](https://github.com/Automattic/woocommerce-payments/pull/4361#issuecomment-1162790138) and I see you've moved the status on the project board to _Review in progress_.

**Comment by james-allan** [comment]
Yeah I'm still happy with this. 👍

**Comment by Jinksi** [comment]
I missed multi-currency support for this filter 👎

I'll get an issue & PR ASAP.

image

**Pull request merged**